### PR TITLE
fix: collect outputs parameters

### PR DESCRIPTION
**Pull request by Tuditi**
# Description of change

- Renamed the argument for `collectOutput`
- Changed the return type of `prepareOutput`
- Changed the type of `Time`
- Added an example to test `collectOutput`

## Links to any relevant issues

https://github.com/iotaledger/firefly/issues/3519

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/nodejs/examples/21-collect-outputs.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+async function handleNewOutput(err, data) {
+    setTimeout(async () => {
+        console.log('Output received:', data)
+        const event = JSON.parse(data)
+        const outputId = event.event.NewOutput.output.outputId
+        console.log(outputId)
+        const resp = await bob.collectOutputs([outputId])
+        console.log('Output has been collected in the following transaction:', resp)
+        process.exit(0)
+    }, 15000)
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/21-collect-outputs.js]: I think this timeout here is useless

File: bindings/nodejs/examples/21-collect-outputs.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                unlocks: {
+                    expiration: {
+                        unixTime: new Date().getSeconds() + 15000
+                    }
+                }
+            },
+        );
+
+        const resp = await alice.sendOutputs([outputData])
+        console.log('Transaction is sent', resp)
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/21-collect-outputs.js:34]: I think you need something like
```suggestion
        console.log('Transaction is sent', resp)
        // Sync bob account after 10 seonconds, so it detects the new output and the event gets emitted
        setTimeout(() => {
            await bob.sync()
        }, 10000);
```

File: bindings/nodejs/examples/21-collect-outputs.js (hunk 1 above)
Comment (Tuditi) [reply bindings/nodejs/examples/21-collect-outputs.js]: Don't we have to wait until the transaction gets confirmed on the Tangle?

File: bindings/nodejs/examples/21-collect-outputs.js (hunk 2 above)
Comment (Tuditi) [reply bindings/nodejs/examples/21-collect-outputs.js:34]: Added that one in the event handler.

File: bindings/nodejs/examples/21-collect-outputs.js (hunk 1 above)
Comment (Thoralf-M) [reply bindings/nodejs/examples/21-collect-outputs.js]: Yes, but when you get the event, then it's already confirmed

File: bindings/nodejs/examples/21-collect-outputs.js (hunk 2 above)
Comment (Thoralf-M) [reply bindings/nodejs/examples/21-collect-outputs.js:34]: But syncing needs to be called to get the event in the first place

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**